sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.http.rest.PagedFlux;
+import reactor.core.publisher.Mono;
+
+import java.util.Map;
+
+/**
+ * class for the table async client
+ */
+@ServiceClient(
+    builder = TableClientBuilder.class,
+    isAsync = true)
+public class TableAsyncClient {
+    private final String tableName;
+
+    TableAsyncClient(String tableName) {
+        this.tableName = tableName;
+    }
+
+    /**
+     * Queries and returns entities in the given table using the select and filter strings
+     *
+     * @param queryOptions the odata query object
+     * @return a paged flux of all the entity which fit this criteria
+     */
+    public PagedFlux<TableEntity> queryEntity(QueryOptions queryOptions) {
```

Comment:
added in a code snippet, let me know if that solution works or if I should talk with the team about adding a exists method. Thanks

Reply:
Thanks for adding it to the code snippet. It might probably be something we should include in the public API itself but that's arch board discussion. For now, we can just keep it in the samples.

Reply:
Also, should the name of the method be `queryEntities()` as it's returning a paged collection?

Reply:
yep just switched the wording. Also we just spoke yesterday in a meeting and we are adding a get() method. This could help resolve your previous concern about exists() since it could be used to test existence.